lib/internal/url.js

```diff
@@ -487,7 +487,7 @@ Object.defineProperties(URL.prototype, {
       if (!search) {
         ctx.query = null;
         ctx.flags &= ~binding.URL_FLAGS_HAS_QUERY;
-        this[searchParams][searchParams] = {};
+        this[searchParams][searchParams] = [];
```

Comment:
Another way to do it is to not return here, create an else branch to wrap the stuff below and leave the initialization to `initSearchParams(this[searchParams], search)` later. That way URL doesn't need to know about the data sturcture of `URLSearchParams`